### DISCUSSION TITLE
Composite types and optional lists in schema models with postgreSQL | CockroachDB

### DISCUSSION THREAD

**Discussion by praveen-palanisamy** [prisma/prisma#18891] [Q&A]
1. Are composite types in schema models like this one really not supported for postgreSQL, MySQL and CockroachDB or am I missing something basic?
![image](https://user-images.githubusercontent.com/4770482/233882624-d02c7c1e-5d21-4dba-98e2-63ddf3892bc2.png)

2. Also, how can we use optional list fields supported by PostgreSQL and CockroachDB since Prisma data model schema doesn't seem to support it?

**Comment by ludralph** [comment]
Hi @praveen-palanisamy 👋 

1. Composite types are only supported in MongoDB as can be seen in the [documentation](https://www.prisma.io/docs/concepts/components/prisma-client/composite-types)

2. Optional list are not supported as can be seen in the [documentation](https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#type-modifiers). There is a [feature request](https://github.com/prisma/prisma/issues/14390) for this. Also, you can look at this Stackoverflow [answer](https://stackoverflow.com/questions/70682378/how-to-create-a-optional-list-field-in-prisma/70683053#70683053) on how to pass an empty value to a list that is not defined as optional.

**Reply by ludralph** [reply]
Hi @praveen-palanisamy 

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
